lib/modules/test.go

```diff
@@ -62,6 +62,7 @@ type TestModules struct {
 func SetTestModules(t *testing.T, testModules *TestModules) {
 	defaultModules := GetModules()
 	t.Cleanup(func() { SetModules(defaultModules) })
+	t.Setenv("TELEPORT_TEST_NOT_SAFE_FOR_PARALLEL", "true")
```

Comment:
can we add a check that verifies if GetModules doesn't have the default content? It would be a stronger check than this one.

cc @zmb3

Reply:
That wouldn't verify whether or not we are currently in a parallel test, it would only verify whether the modules were previously changed, which is technically something that's allowed in a non-parallel test. This would be a weaker check IMO.

Go's testing package doesn't expose an API to tell you whether or not you are in a parallel test, but they do have those checks internally, and `t.Setenv` will panic if called in a parallel test.

With this in place, it would not have been possible to merge the bad code in the first place.